src/migrations/20240420000000-correct-onApprovedAR-values.js

```diff
@@ -0,0 +1,252 @@
+const {
+  prepMigration,
+} = require('../lib/migration');
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      await prepMigration(queryInterface, transaction, __filename);
+      await queryInterface.sequelize.query(/* sql */`
+      --  1. Calculate correct onApprovedAR values for objectives
+      DROP TABLE IF EXISTS objectives_on_approved_ars;
+      CREATE TEMP TABLE objectives_on_approved_ars
+      AS
+      SELECT
+        o.id oid,
+        BOOL_OR(ar.id IS NOT NULL) on_approved_ar
+      FROM "Objectives" o
+      LEFT JOIN "ActivityReportObjectives" aro
+        ON o.id = aro."objectiveId"
+      LEFT JOIN "ActivityReports" ar
+        ON aro."activityReportId" = ar.id
+        AND ar."calculatedStatus" = 'approved'
+      GROUP BY 1
+      ;
+      --  1. Calculate correct onApprovedAR values for goals
+      DROP TABLE IF EXISTS goals_on_approved_ars;
+      CREATE TEMP TABLE goals_on_approved_ars
+      AS
+      SELECT
+        g.id gid,
+        BOOL_OR(ar.id IS NOT NULL OR COALESCE(ooaa.on_approved_ar,FALSE)) on_approved_ar
+      FROM "Goals" g
+      LEFT JOIN "ActivityReportGoals" arg
+        ON g.id = arg."goalId"
+      LEFT JOIN "ActivityReports" ar
+        ON arg."activityReportId" = ar.id
+        AND ar."calculatedStatus" = 'approved'
+      LEFT JOIN "Objectives" o
+        ON o."goalId" = g.id
+      LEFT JOIN objectives_on_approved_ars ooaa
+        ON ooaa.oid = o.id
+      GROUP BY 1
+      ;
+      DROP TABLE IF EXISTS initial_objective_stats;
+      CREATE TEMP TABLE initial_objective_stats
+      AS
+      SELECT
+        SUM(CASE WHEN on_approved_ar = "onApprovedAR" THEN 1 ELSE 0 END
+        ) matching_values_o,
+        SUM(
+          CASE
+            WHEN on_approved_ar = "onApprovedAR"
+              OR "onApprovedAR" IS NULL
+          THEN 0
+          ELSE 1
+          END
+        ) incorrect_values_o,
```

Comment:
while this does work it takes a bit more time to read and understand what is being done here. `SUM(CASE)` is also a bit slower then the alternative `count() filter (where)`.

Reply:
It was complaining for some reason and I didn't feel like figuring out why. Though now that I think about it I think I know